rust/publish-zenodo/src/lib.rs

```diff
@@ -0,0 +1,575 @@
+// TODO: use the [Research Organization Registry (ROR)](https://ror.org/) database to gather affiliations for authors
+
+
+use std::{path::PathBuf, str::FromStr, sync::{Arc, Mutex}};
+
+use codec::{schema::{Primitive, PropertyValue, PropertyValueOrString}, Codec};
+use cli_utils::parse_host;
+use common::{
+    clap::{self, Parser}, eyre::{bail, OptionExt, Result}, reqwest::Client, serde, serde_json::{json, Value}, tempfile, tokio, tracing,
+};
+
+use document::{schema, CommandWait, Document, EncodeOptions, Format, schema::Node};
+use codec_swb::SwbCodec;
+
+mod metadata_extraction;
+
+fn parse_date(input: &str) -> Result<schema::Date> {
+    Ok(schema::Date::from_str(input)?)
+}
+
+
+/// Items within Zenodo's controlled vocabulary of accepted types of publication
+#[derive(Debug, Clone, Copy, Default, serde::Deserialize, serde::Serialize, clap::ValueEnum)]
+#[serde(crate = "common::serde", rename_all = "lowercase")]
+enum PublicationType {
+    AnnotationCollection,
+    Book,
+    Section,
+    ConferencePaper,
+    DataManagementPlan,
+    Article,
+    Patent,
+    #[default]
+    Preprint,
+    Deliverable,
+    Milestone,
+    Proposal,
+    Report,
+    SoftwareDocumentation,
+    TaxonomicTreatment,
+    TechnicalNote,
+    Thesis,
+    WorkingPaper,
+    Other,
+}
+
+/// Publish to Zenodo
+/// 
+/// Metadata for the deposition is extracted from the metadata within the
+/// document.
+/// 
+/// Once uploaded, you will prompted to review the deposition and publish from
+/// Zenodo's web interface. If you wish to skip the review process and publish
+/// immediately, then use the `--force` flag.
+/// 
+/// By default, Zenodo's testing server is used. This means that DOIs created
+/// will not resolve. To upload to Zenodo's public-facing production server, use
+/// the `--zenodo` flag. To upload to a specific server instance, use
+/// `--server`.
+/// 
+/// Unless told otherwise, your deposit will be given the publication type "preprint".
```

Comment:
Add short message with guidance on how to change this from `preprint` to other alternatives.